core/src/main/java/io/stargate/core/util/ByteBufferUtils.java

```diff
@@ -31,6 +31,22 @@ public static ByteBuffer fromBase64(String base64) {
     return ByteBuffer.wrap(Base64.getDecoder().decode(base64));
   }
 
+  public static String toBase64ForUrl(ByteBuffer buffer) {
+    return Base64.getUrlEncoder().encodeToString(getArray(buffer));
+  }
+
+  public static ByteBuffer fromBase64UrlParam(String base64) {
```

Comment:
Few questions here:

1. Is this only to be backwards-compatible with page state already returned? 
2. So how do you know what's the diff between the base encoder and the url encoder? Why searching only for these char and why this single replace?

Going a bit deeper and looking into this (https://datatracker.ietf.org/doc/html/rfc4648#section-4), seems that the diff in encoding is not what you have here? 

```
This encoding is technically identical to the previous one, except
   for the 62:nd and 63:rd alphabet character, as indicated in Table 2.
```   

Maybe I am wrong, but let's double check on this.

Reply:
Yea so `/` should also be checked for I guess, as it can be returned in the base encoding, same as the `+`. Url encoded value is `%2F`.

Reply:
> Is this only to be backwards-compatible with page state already returned?

backward-compatibility is handled inside the `if` on line 40.

Line 47 is the regular use case (transparent to URL encoding).

> So how do you know what's the diff between the base encoder and the url encoder? Why searching only for these char and why this single replace?

Replacing ` ` (space) chars is to fix older base64 text blocks that get broken at the HTTP level where `+` gets replaces with ` ` during URL parameter decoding. Since the base64 alphabet does not use spaces, this transformation should be safe as far as I can tell.

If neither `/` nor `+` are in the base64 encoded string, then the "URL decoder" should be able to decode the string correctly, I believe, even if it was produced by the "plain" encoder.

Do you see a particular failure mode that is not handled by this code?

Reply:
My bad, resolving.